src/operator/nn/softmax-inl.h

```diff
@@ -301,7 +282,7 @@ __global__ void softmax_compute_kernel(DType *in, OType *out, index_t M, int axi
 
   red::sum::SetInitValue(smem[x]);
```

Comment:
Does have multiple max values affect numerical accuracy? Or are they reduced at some other point to a final max?

Reply:
Not sure I understand.
![softmax](https://user-images.githubusercontent.com/8398980/63448504-e0a62e80-c3f2-11e9-8b57-1382384f07e5.gif)

That is not very numerically stable, so there is another step introduced that finds the maximum x_i first and compute
![softmax](https://user-images.githubusercontent.com/8398980/63448793-92455f80-c3f3-11e9-8016-ed54e2289e69.gif)
It does not matter which maximum you take.

Reply:
let me rephrase. When running Reduce1D, are all the max values reduce into smem[0] / smax? as I understand the xmxax should be the max of {x_i}. It actually matters for numerical issues.

Reply:
Yes, those intermediate max values are then reduced across threads to the final maximum.